### PR TITLE
Manage course visibility to anonymous users

### PR DESCRIPTION
**Pull request by HamzaIbnFarooq**
### Story Link
https://edlyio.atlassian.net/browse/EDE-322
### Description
Restrict anonymous users from viewing the courses and their about page.

This PR allows admin to achieve this using **Site Configurations**. The admin needs to add
**'SHOW_COURSES_TO_ANONYMOUS_USERS' : false** in that specific site's configurations to turn this feature on or off.

The default value of **SHOW_COURSES_TO_ANONYMOUS_USERS** is **False**.

### CODE REVIEW

File: lms/envs/common.py [hunk 1]
```diff
@@ -2963,6 +2963,10 @@ def _make_locale_paths(settings):
 # visible. We default this to the legacy permission 'see_exists'.
 COURSE_ABOUT_VISIBILITY_PERMISSION = 'see_exists'
 
+# [COLARAZ_CUSTOM]
+# it will restrict anonymous users from viewing courses on index, courses and course about pages
+SHOW_COURSES_TO_ANONYMOUS_USERS = False
```
Comment (tasawernawaz) [inline lms/envs/common.py]: ca we add a unit test for this setting? (there is not later :P) also please make sure to ask DevOps about this flag, so they can add this in the config var.

Comment (danialmalik) [inline lms/envs/common.py]: Please add this config in `production.py` too so that this could be configured through the configurations later.

Comment (HamzaIbnFarooq) [reply lms/envs/common.py]: Sorry, couldn't understand your point.

Comment (HamzaIbnFarooq) [reply lms/envs/common.py]: Yes, I will get that variable added once you approve this PR.
Secondly, a lot of test cases need to be written for Colaraz's edx-platform, don't you think we need to have a meeting with them and get extra for writing tests?

Comment (danialmalik) [reply lms/envs/common.py]: This is a new variable. right?
If we want it to be configured through DevOps configurations i.e. through `lms.env.json` file, we have to add this in production.py in the following way:
```
SHOW_COURSES_TO_ANONYMOUS_USERS = ENV_TOKENS.get('SHOW_COURSES_TO_ANONYMOUS_USERS', SHOW_COURSES_TO_ANONYMOUS_USERS)
```

Comment (HamzaIbnFarooq) [reply lms/envs/common.py]: As this variable will be used in Site configurations so I don't think there is a need for that but still am going to add it now.

Comment (tasawernawaz) [reply lms/envs/common.py]: sad but true :) I think we should give estimates including effort for writing tests and tests should be part of the changes.

Comment (danialmalik) [reply lms/envs/common.py]: SInce we have it in `common.py`, it would be better to have it there too. 
It's always better to have more options :D

Comment (HamzaIbnFarooq) [reply lms/envs/common.py]: Not always...

Comment (HamzaIbnFarooq) [reply lms/envs/common.py]: done

**Pull request merged**